### PR TITLE
Compress archived files

### PR DESCRIPTION
**Pull request by etiennehomer**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
No

**What kind of change does this PR introduce?**

**What is the current behavior?**
Archived files are exploded in S3 as plain files.

**What is the new behavior (if this is a feature change)?**
Archived files are exploded in S3 and they are compressed as gzip files.

### CODE REVIEW

File: src/main/java/com/powsybl/caseserver/datasource/S3CaseDataSourceService.java [hunk 1]
```diff
@@ -51,34 +53,30 @@ public Boolean datasourceExists(UUID caseUuid, String fileName) {
...
+        String caseFileKey;
+        // For archived cases (.zip, .tar, ...), individual files are gzipped in S3 server.
+        // Here the requested file is decompressed and simply returned.
+        if (S3CaseService.isArchivedCaseFile(caseName)) {
+            caseFileKey = uuidToKeyWithFileName(caseUuid, fileName + GZIP_EXTENSION);
+            return s3CaseService.withS3DownloadedTempPath(caseUuid, caseFileKey,
+                    file -> S3CaseService.decompress(Files.readAllBytes(file)));
+        }
+        caseFileKey = uuidToKeyWithFileName(caseUuid, caseName);
         return withS3DownloadedDataSource(caseUuid, caseFileKey,
```
Comment (jonenst) [inline src/main/java/com/powsybl/caseserver/datasource/S3CaseDataSourceService.java:66]: should use withS3DownloadedTempPath as well ?

File: src/main/java/com/powsybl/caseserver/datasource/S3CaseDataSourceService.java [hunk 2]
```diff
@@ -51,34 +53,30 @@ public Boolean datasourceExists(UUID caseUuid, String fileName) {
...
         return s3CaseService.listName(caseUuid, regex);
     }
 
-    public <R, T extends Throwable> R withS3DownloadedDataSource(UUID caseUuid, FailableFunction<DataSource, R, T> f) {
-        FailableFunction<Path, DataSource, T> pathToDataSource = DataSource::fromPath;
-        FailableFunction<Path, R, T> composedFunction = pathToDataSource.andThen(f);
-        return s3CaseService.withS3DownloadedTempPath(caseUuid, composedFunction);
-    }
-
     public <R, T extends Throwable> R withS3DownloadedDataSource(UUID caseUuid, String caseFileKey, FailableFunction<DataSource, R, T> f) {
```
Comment (jonenst) [inline src/main/java/com/powsybl/caseserver/datasource/S3CaseDataSourceService.java:80]: not sure if this one is needed anymore, we should never download the whole datasource now

File: src/main/java/com/powsybl/caseserver/datasource/S3CaseDataSourceService.java (hunk 1 above)
Comment (etiennehomer) [reply src/main/java/com/powsybl/caseserver/datasource/S3CaseDataSourceService.java]: Yes both withS3DownloadedDataSource can be deleted

File: src/main/java/com/powsybl/caseserver/datasource/S3CaseDataSourceService.java (hunk 2 above)
Comment (etiennehomer) [reply src/main/java/com/powsybl/caseserver/datasource/S3CaseDataSourceService.java]: Yes both withS3DownloadedDataSource can be deleted

**Pull request merged**